### DISCUSSION TITLE
Distinguish the types of notes in the osu!taiko editor timeline

### DISCUSSION THREAD

**Discussion by Hiviexd** [ppy/osu#13435] [Ideas] [answered]
Currently, stable shows the color of a note in the timeline, and most taiko mappers rely on that to map and mod.

The issue with lazer is that the notes share the same white color in the timeline, which makes mapping feel extremely offputting and weird. We also rely on that in order to look at the map from a wider perspective using the zoom function.

I propose to show note colors on the timeline, which would make the taiko mapping experience much more intuitive and familiar.

### what stable looks like:
![stable](https://user-images.githubusercontent.com/62819481/121578697-da979c80-ca22-11eb-8465-cf7e9d75b8a9.png)

### what lazer looks like:
![lazer](https://user-images.githubusercontent.com/62819481/121578705-dd928d00-ca22-11eb-992a-78a37d9237b1.png)

### proposed change:
![example](https://user-images.githubusercontent.com/62819481/121578716-e08d7d80-ca22-11eb-9e60-943d85ba4844.png)

Oh also I don't know how possible is this but in addition to this, showing finishers on the timeline would be a direct upgrade from the stable editor, instead of just aiming for feature parity with this proposal.

**Comment by Heaxys** [comment]
being able to see if a note is a finisher should already have been a normal thing for a long time, and not seeing colors directly in the timeline is just peeing into mapper's mouths to be honest, it's literally mapping without having any visual of your map

**Comment by peppy** [comment, marked as answer]
Tracking via issue https://github.com/ppy/osu/issues/13442.

**Reply by Hiviexd** [reply]
I just noticed that drumrolls and swells look exactly the same in the timeline after this update, any chance you could look into it? or should I open a separate issue?
![image](https://user-images.githubusercontent.com/62819481/121788807-4109ef80-cbc8-11eb-9076-8d2b716ff10c.png)

**Reply by peppy** [reply]
That goes for all long objects right now, including spinners and mania holds. It's a lower priority though. You can already see the difference on the playfield below, which should be how you are mapping.
